### ISSUE TITLE
No `Hot Reload` when importing TailwindConfig from another `tailwind.config.ts` file

### ISSUE THREAD

**Issue by sshmaxime** [tailwindlabs/tailwindcss#13419]
**What version of Tailwind CSS are you using?**

**3.3**

**What build tool (or framework if it abstracts the build tool) are you using?**

**Next: 14.1.0**
**Postcss: 8**

**What version of Node.js are you using?**

**20.11.0**

**What browser are you using?**

**Chrome**

**What operating system are you using?**

**MacOS**

**Reproduction URL**

[Codesandbox](https://codesandbox.io/p/github/dan5py/turborepo-shadcn-ui/csb-3hx9l8/draft/cranky-smoke?file=%2Fapps%2Fdocs%2Fapp%2Fpage.tsx%3A11%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clugibfd800073b6jx8nfrfx7%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clugibfd800023b6jz0fhb4xx%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clugibfd800043b6jwxrt3s2p%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clugibfd800063b6jzhen1f8y%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clugibfd800023b6jz0fhb4xx%2522%253A%257B%2522id%2522%253A%2522clugibfd800023b6jz0fhb4xx%2522%252C%2522activeTabId%2522%253A%2522clugiei0t00463b6jal7hu3th%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clugidn0d00023b6jtmw1hiec%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A11%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A11%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fapps%252Fdocs%252Fapp%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clugiei0t00463b6jal7hu3th%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clugibfd800063b6jzhen1f8y%2522%253A%257B%2522id%2522%253A%2522clugibfd800063b6jzhen1f8y%2522%252C%2522activeTabId%2522%253A%2522clugibfd800053b6jqy6b37ho%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clugibfd800053b6jqy6b37ho%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3001%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3001%252C%2522id%2522%253A%2522clugifemm00dd3b6j5bc6ndqn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clugibfd800043b6jwxrt3s2p%2522%253A%257B%2522id%2522%253A%2522clugibfd800043b6jwxrt3s2p%2522%252C%2522activeTabId%2522%253A%2522clugiiybh01bx3b6jdkct0yc2%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clugiiybh01bx3b6jdkct0yc2%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Describe your issue**

I'm using a monorepo and have my `tailwind.config.ts` in a `ui` package that I import from my Nextjs apps like so `export * from "@repo/ui/tailwind.config";` in my apps `tailwind.config.ts`.

Basically, what I expect is that when I change something in that file it `hot reload` like it does in a normal setup. But it doesn't hot reload and I have to shutdown dev and reload it manually.

Thanks for your help 🙏

**Comment by ryan1969manus** [comment]
Can you tell me your query is solve or not?  [CoveredCalifornia](https://www.mycoveredcalifornia.com)

**Comment by KOCNS1** [comment]
I believe that simply importing your desired config from another folder within your monorepo is not the correct approach and is not possible. However, you can import your main Tailwind CSS config into your app if you include a tailwind.config.ts in your app's root and use the presets option to import the main config.

Example:

```js
// This is the config file that should be at the root of your app

/** @type {import('tailwindcss').Config} */
module.exports = {
  presets: [
    // Assuming you have a path alias for your UI library; otherwise, put the relative path here
    require('@ui/shared/tailwind.config')
  ],
  // ...
}
```

by doing so, and assuming that the `content` option is filled correctly, Hot Reload should work as expected

[Here's the documentation](https://tailwindcss.com/docs/presets) if you want more information about it 😄

**Comment by amanda2369weaver** [comment]
Thank you for this information. [Official Site](https://www.my-sedgwick.com)

**Comment by philipp-spiess** [comment]
Hey! So the way hot-reloading for imported JavaScript modules works is that if the imported package looks like an npm module, it won't include it in the lists of files to watch for changes. This is a heuristics we've added to improve performance for most use cases since we expect npm package installs/updates to require a reload. This should work if you change the import to a relative path instead!

Going to close this for now as we still think that's a good tradeoff and we've done the same for v4 for now. We're happy to reconsider alternatives to this heuristics in the future though. Thanks!
